Sources/AWSLambdaEvents/APIGateway+V2.swift

```diff
@@ -117,3 +119,79 @@ extension APIGateway.V2 {
         }
     }
 }
+
+// MARK: - Codable Request body
+
+extension APIGateway.V2.Request {
+    /// Generic body decoder for JSON payloads
+    ///
+    /// Example:
+    /// ```
+    /// struct Request: Codable {
+    ///   let value: String
+    /// }
+    ///
+    /// func handle(context: Context, event: APIGateway.V2.Request, callback: @escaping (Result<APIGateway.V2.Response, Error>) -> Void) {
+    ///   do {
+    ///     let request: Request? = try event.decodedBody()
+    ///     // Do something with `request`
+    ///     callback(.success(APIGateway.V2.Response(statusCode: .ok, body:"")))
+    ///   }
+    ///   catch {
+    ///     callback(.failure(error))
+    ///   }
+    /// }
+    /// ```
+    ///
+    /// - Throws: `DecodingError` if body contains a value that couldn't be decoded
+    /// - Returns: Decoded payload. Returns `nil` if body property is `nil`.
+    public func decodedBody<Payload: Codable>() throws -> Payload? {
+        guard let bodyString = body else {
+            return nil
+        }
+        let data = Data(bodyString.utf8)
+        return try JSONDecoder().decode(Payload.self, from: data)
```

Comment:
how about:
`public func decodedBody<Payload: Codable>(decoder:JSONDecoder = JSONDecoder()) throws -> Payload? {` 

This way the user has can customize the decoder and provide their own. For example, to change the dateDecodingStrategy.

Reply:
(Same suggestion for the JSONEncoder usage below)

Reply:
That's a good point but also yet another instance where "TopLevelDecoder" types would be the right thing to go with... Since we'd like to allow using various coder implementations, not marry the API to Foundation's impl only.

We could do them ad-hoc here in the lambda lib again as Combine does, but that's growing the problem a bit; Or we ignore and go along with this for now, but eventually try to resolve it once we get to it.

// fyi @tomerd

Reply:
Thanks for the feedback. I've updated the PR to inject both `JSONEncoder` and `JSONDecoder`.

I agree a "TopLevelDecoder" type would be ideal, let me know if I should do any further changes.

Reply:
Another idea would be to see if it's possible to push the functions out into an "AWSLambdaFoundationCompat" module or something similar?

Reply:
@ktoso I was looking at `Lambda+Codable.swift` from `AWSLambdaRuntime` and it seems like encoding/decoding work could be moved to a separate module, like you suggest.

Would you be open for that work being done on a separate pull request? While I think it is valuable, it feels out of the initial scope of this work.

Reply:
Sorry, I don't know the internals here, based on a github search I don't see too many instances of "import Foundation". Would that mean that the lambda runtime wouldn't depend on `Foundation`? That might be nice for performance, right?

Reply:
@CalCohen, at the moment, yes, `AWSLambdaRuntime` depends on `Foundation` for `Codable` support:

https://github.com/swift-server/swift-aws-lambda-runtime/blob/246088e340bd3515b93358ef6b39cec0dc80618b/Sources/AWSLambdaRuntime/Lambda%2BCodable.swift#L17